src/CommunityToolkit.Maui.Core/Essentials/SaveFileDialog/SaveFileDialogImplementation.macios.cs

```diff
@@ -0,0 +1,56 @@
+namespace CommunityToolkit.Maui.Storage;
+
+/// <inheritdoc cref="ISaveFileDialog" />
+sealed partial class SaveFileDialogImplementation : ISaveFileDialog, IDisposable
+{
+	UIDocumentPickerViewController? documentPickerViewController;
+	TaskCompletionSource<string>? taskCompetedSource;
+	
+	/// <inheritdoc/>
+	public async Task<string> SaveAsync(string initialPath, string fileName, Stream stream, CancellationToken cancellationToken)
+	{
+		cancellationToken.ThrowIfCancellationRequested();
+		var fileManager = NSFileManager.DefaultManager;
+		var fileUrl = fileManager.GetTemporaryDirectory().Append($"{Guid.NewGuid()}{GetExtension(fileName)}", false);
+		await WriteStream(stream, fileUrl.Path ?? throw new Exception("Path cannot be null."), cancellationToken);
+
+		cancellationToken.ThrowIfCancellationRequested();
+		taskCompetedSource = new TaskCompletionSource<string>();
+		
+		documentPickerViewController = new UIDocumentPickerViewController(new[] { fileUrl });
+		documentPickerViewController.DidPickDocumentAtUrls += DocumentPickerViewControllerOnDidPickDocumentAtUrls;
+		documentPickerViewController.WasCancelled += DocumentPickerViewControllerOnWasCancelled;
+		
+		var currentViewController = Microsoft.Maui.Platform.UIApplicationExtensions.GetKeyWindow(UIApplication.SharedApplication)?.RootViewController;
+		currentViewController?.PresentViewController(documentPickerViewController, true, null);
+
+		return await taskCompetedSource.Task.WaitAsync(cancellationToken).ConfigureAwait(false);
+	}
+
+	/// <inheritdoc/>
+	public Task<string> SaveAsync(string fileName, Stream stream, CancellationToken cancellationToken)
+	{
+		return SaveAsync("/", fileName, stream, cancellationToken);
+	}
+
+	/// <inheritdoc />
+	public void Dispose()
```

Comment:
who calls the Dispose? If it's called when the service is disposed those events will be subscribed to more than once.